File: Practice/06/Python/06.py
import math

a = int(input("Введите a = "))
b = int(input("Введите b = "))
c = int(input("Введите c = "))


if a == 0 and b == 0:
    print("x - Действительные числа") if c == 0 else print("Решений нет") 
elif a == 0:
    print(f"x = {-c/b}")
else:
    d = b * b - 4 * a * c
    if d < 0:
        print(f"Нет корней среди действительных чисел ( d == {d})")
    elif d == 0:
        print(f"x = {-b / (2 * a)}")
    else:
        print(f"x1 = {(-b + math.sqrt(d)) / (2 * a)}, x2 = {(-b - math.sqrt(d)) / (2 * a)}")

File: Practice/06/C++/06.cpp
#include <iostream>
#include <math.h>

using namespace std;

template <typename Num>

Num input(string msg)
{
    Num var;
    cout << msg;
    cin >> var;
    cout << endl;
    return var;
}

int main()
{
    double a = input<double>("Input a = ");
    double b = input<double>("Input b = ");
    double c = input<double>("Input c = ");

    if (a == 0 && b == 0)
    {
        cout << "x = " << -c << endl;
    } else if (a == 0)
    {
        cout << "x = " << -c/b << endl;
    }
    else {
        double d = b * b - 4 * a * c;
        if (d  < 0)
        {
            cout << "Does not exist in real numbers";
        } else if (d == 0)
        {
            cout << "x = " << -b / (2 * a);
        } else 
        {
            cout << "x1 = " << (-b + sqrt(d)) / (2 * a)
                << " x2 = " << (-b - sqrt(d)) / (2 * a)
                << endl;
        }
    }
    
    
}